### PR TITLE
Issue 40: Updates for drugFromSource to support drug combinations

### LINKED ISSUE #40: Updates for drugFromSource and drugFromSourceId to support drug combinations
As discussed in [22 April meeting](https://docs.google.com/document/d/1_ayBXr2Q-4z0nxCpCi5L7SfzoDi0OPsp_KIBBRYe7vI/edit#heading=h.8jl5utc0yth3), we should make the attribute `drugFromSource` a list when split on `/` only (other delimiters have a different semantic in PGKB and should continue to be exploded into separate evidence) and rename the attribute to be plural `drugsFromSource`.

This is to support drug combinations like [ivacaftor / lumacaftor](https://www.pharmgkb.org/chemical/PA166152935).

To remove complexity and because CHEBI IDs aren't being used by OT, we will remove the `drugFromSourceId` attribute entirely.

Related OT issue [here](https://github.com/opentargets/issues/issues/3205).

### PR DESCRIPTION
**Pull request by apriltuesday**
Closes #40. Tests won't pass until the OT PGx schema is updated, but [here](https://github.com/EBIvariation/opentargets-pharmgkb/actions/runs/8818518918/job/24207491022) is a test run with validation commented out.

Git struggles with the expected test output diff, [this](https://www.diffchecker.com/9S4ud6ok/) is slightly better. Example evidence string from the test:
```
{
  "datasourceId": "pharmgkb",
  "datasourceVersion": "2023-03-23",
  "datatypeId": "clinical_annotation",
  "studyId": "1447979749",
  "evidenceLevel": "1A",
  "literature": ["29126871", "28606620", "27898234", "27898234", "25981758", "27805836", "27805836", "29327948", "29451946", "28325531", "27334259", "24973281", "24973281", "24973281", "27298017"],
  "genotype": "CTT/del",
  "genotypeAnnotationText": "Patients with cystic fibrosis and the rs113993960 CTT/del genotype (one copy of the CFTR F508del variant) may experience a limited benefit from treatment with the combination drug of ivacaftor/lumacaftor, as shown by improvement in sweat chloride concentrations CFQ-R questionnaire scores when compared to treatment with placebo. However, ppFEV1, BMI or body weight did not show a significant improvement following ivacaftor/lumacaftor treatment. This genotype is not an indication for use of the combination drug of ivacaftor/lumacaftor according to the FDA-approved drug label for this drug combination. Other genetic and clinical factors may also influence response to ivacaftor/lumacaftor.",
  "drugsFromSource": [
    "ivacaftor",
    "lumacaftor"
  ],
  "pgxCategory": "efficacy",
  "phenotypeText": "Cystic Fibrosis",
  "phenotypeFromSourceId": "MONDO_0009061",
  "genotypeId": "21_5010011_CCTT_C,CCTT",
  "variantRsId": "rs113993960"
}
```

FYI @DSuveges @ireneisdoomed

### CODE REVIEW

File: opentargets_pharmgkb/pandas_utils.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def none_to_nan(x):
     return np.nan if x is None else x
 
 
-def explode_column(df, source_col, target_col, sep=';'):
+def explode_column(df, source_col, target_col, sep=';', split_only=False):
```
Comment (tcezard) [inline opentargets_pharmgkb/pandas_utils.py:13]: I'm being a bit picky here but running `explode_column` without exploding seems odd.

File: opentargets_pharmgkb/evidence_generation.py [hunk 2]
```diff
@@ -296,38 +293,18 @@ def explode_and_map_genes(df):
...
-    mapped_drugs = pd.merge(mapped_drugs, drugs_table, left_on='split_drug', right_on='Name', how='left')
-    mapped_drugs['chebi'].fillna(mapped_drugs['chebi_id'].apply(chebi_id_to_iri), inplace=True)
-    return mapped_drugs
-
-
-def chebi_id_to_iri(id_):
-    if pd.notna(id_):
-        return f'http://purl.obolibrary.org/obo/CHEBI_{id_}'
-    return None
+    # Drugs explicitly annotated as combinations
```
Comment (tcezard) [inline opentargets_pharmgkb/evidence_generation.py:305]:
```suggestion
    # Drugs explicitly annotated as combinations are kept as a list of drug names
```

File: opentargets_pharmgkb/pandas_utils.py (hunk 1 above)
Comment (tcezard) [reply opentargets_pharmgkb/pandas_utils.py:13]: Or is it that `explode_column` means split column and it can be split into a list or into many rows.

**Review by tcezard — APPROVED** [review]

File: opentargets_pharmgkb/pandas_utils.py (hunk 1 above)
Comment (apriltuesday) [reply opentargets_pharmgkb/pandas_utils.py:13]: Yeah this kind of bothered me too... what about naming it `split_and_explode_column`? This way `split` == split the string into a list and `explode` == explode the list value into rows

Comment (tcezard) [reply opentargets_pharmgkb/pandas_utils.py:13]: Sounds reasonable

**Review by tcezard — APPROVED** [review]

**Pull request merged**